aten/src/ATen/native/cudnn/Conv_v8.cpp

```diff
@@ -121,56 +115,125 @@ void raw_cudnn_convolution_forward_out(
+
+  CacheKey key;
+  get_cachekey(key, CUDNN_BACKEND_OPERATION_CONVOLUTION_FORWARD_DESCRIPTOR, output, input, weight, padding, stride, dilation, groups, deterministic, allow_tf32);
+  auto search = engine_cache.find(key);
+  if (search != engine_cache.end()) {
+    run_conv_cfg(handle, input, output, weight, search->second);
+    return;
+  }
+
+  cudnn_frontend::EngineConfigList filtered_configs;
+  get_configs(filtered_configs, handle, CUDNN_BACKEND_OPERATION_CONVOLUTION_FORWARD_DESCRIPTOR, input, output, weight, key, padding, stride, dilation, deterministic, allow_tf32);
+
+  try_filtered_configs(filtered_configs, key, handle, input, output, weight);
+}
+
+void raw_cudnn_convolution_backward_input_out(
+    const at::Tensor& grad_input,
+    const at::Tensor& grad_output,
+    const at::Tensor& weight,
+    IntArrayRef padding, IntArrayRef stride, IntArrayRef dilation, int64_t groups,
+    bool benchmark, bool deterministic, bool allow_tf32) {
+  TORCH_CHECK(!benchmark, "not supported yet");
+  if (grad_input.numel() == 0) {
+    return;
+  }
+  cudnnHandle_t handle = getCudnnHandle();
+
+  CacheKey key;
+  get_cachekey(key, CUDNN_BACKEND_OPERATION_CONVOLUTION_BACKWARD_DATA_DESCRIPTOR, grad_output, grad_input, weight, padding, stride, dilation, groups, deterministic, allow_tf32);
+  auto search = engine_cache.find(key);
+  if (search != engine_cache.end()) {
+    run_conv_cfg(handle, grad_input, grad_output, weight, search->second);
+    return;
+  }
+
+  cudnn_frontend::EngineConfigList filtered_configs;
+  get_configs(filtered_configs, handle, CUDNN_BACKEND_OPERATION_CONVOLUTION_BACKWARD_DATA_DESCRIPTOR, grad_input, grad_output, weight, key, padding, stride, dilation, deterministic, allow_tf32);
+  try_filtered_configs(filtered_configs, key, handle, grad_input, grad_output, weight);
+}
+
+void raw_cudnn_convolution_backward_weight_out(
+    const Tensor& grad_weight, const Tensor& grad_output, const Tensor& input,
+    IntArrayRef padding, IntArrayRef stride, IntArrayRef dilation, int64_t groups,
+    bool benchmark, bool deterministic, bool allow_tf32) {
+  TORCH_CHECK(!benchmark, "not supported yet");
+  if (grad_weight.numel() == 0) {
+    return;
+  }
+  cudnnHandle_t handle = getCudnnHandle();
+
+  CacheKey key;
+  get_cachekey(key, CUDNN_BACKEND_OPERATION_CONVOLUTION_BACKWARD_FILTER_DESCRIPTOR, grad_output, input, grad_weight, padding, stride, dilation, groups, deterministic, allow_tf32);
+
+  auto search = engine_cache.find(key);
+  if (search != engine_cache.end()) {
+    run_conv_cfg(handle, input, grad_output, grad_weight, search->second);
+    return;
+  }
+
+  cudnn_frontend::EngineConfigList filtered_configs;
+  get_configs(filtered_configs, handle, CUDNN_BACKEND_OPERATION_CONVOLUTION_BACKWARD_FILTER_DESCRIPTOR, input, grad_output, grad_weight, key, padding, stride, dilation, deterministic, allow_tf32);
+  try_filtered_configs(filtered_configs, key, handle, input, grad_output, grad_weight);
+}
+
```

Comment:
Look like there are still lots of copy-pasted code here. Could we do something like this:

```C++
// used for convolution and transposed convolution's forward, backward data, and backward filter
void run_single_conv_operation(
    cudnnBackendDescriptorType_t operation,
    const Tensor& y, const Tensor& x, const Tensor& w,
    IntArrayRef padding, IntArrayRef stride, IntArrayRef dilation, int64_t groups,
    bool benchmark, bool deterministic, bool allow_tf32)
{ ... }

void raw_cudnn_convolution_forward_out(
    const Tensor& output, const Tensor& input, const Tensor& weight,
    IntArrayRef padding, IntArrayRef stride, IntArrayRef dilation, int64_t groups,
    bool benchmark, bool deterministic, bool allow_tf32)
{
  run_single_conv_operation(CUDNN_BACKEND_OPERATION_CONVOLUTION_FORWARD_DESCRIPTOR, ...);
}

void raw_cudnn_convolution_backward_input_out(
    const at::Tensor& grad_input,
    const at::Tensor& grad_output,
    const at::Tensor& weight,
    IntArrayRef padding, IntArrayRef stride, IntArrayRef dilation, int64_t groups,
    bool benchmark, bool deterministic, bool allow_tf32)
{
  run_single_conv_operation(CUDNN_BACKEND_OPERATION_CONVOLUTION_BACKWARD_DATA_DESCRIPTOR, ...);
}

void raw_cudnn_convolution_backward_weight_out(
    const Tensor& grad_weight, const Tensor& grad_output, const Tensor& input,
    IntArrayRef padding, IntArrayRef stride, IntArrayRef dilation, int64_t groups,
    bool benchmark, bool deterministic, bool allow_tf32)
{
  run_single_conv_operation(CUDNN_BACKEND_OPERATION_CONVOLUTION_BACKWARD_FILTER_DESCRIPTOR, ...);
}
```